src/components/DetailsModal.tsx

```diff
@@ -0,0 +1,167 @@
+  Alert,
+  Button,
+  Divider,
+  Modal,
+  ModalBody,
+  ModalFooter,
+  Pill,
+  ProgressTracker,
+} from "@us-gov-cdc/cdc-react";
+
+import { Icons } from "@us-gov-cdc/cdc-react-icons";
+import { IFileSubmission } from "@types";
+import getStatusDisplayValuesById, {
+  StatusDisplayValues,
+} from "src/utils/helperFunctions/statusDisplayValues";
+import jsonPrettyPrint from "src/utils/helperFunctions/jsonPrettyPrint";
+import getSubmissionDetails from "src/utils/api/submissionDetails";
+
+interface PropTypes {
+  submission: IFileSubmission;
+  isModalOpen: boolean;
+  handleModalClose: () => void;
+}
+
+function DetailsModal({
+  submission,
+  isModalOpen,
+  handleModalClose,
+}: PropTypes) {
+  const displayValues: StatusDisplayValues = getStatusDisplayValuesById(
+    submission.status
+  );
+  const auth = useAuth();
+  const [details, setDetails] = useState();
+
+  useEffect(() => {
+    const fetchCall = async () => {
+      const res = await getSubmissionDetails(
+        auth.user?.access_token || "",
+        submission.upload_id
+      );
+
+      if (res.status != 200) return;
+
+      try {
+        // TODO: replace any with correct object structure
+        const data = await res.json();
+        setDetails(data);
+      } catch (error) {
+        console.error("Failed to parse JSON:", error);
+      }
+    };
+    fetchCall();
+  }, [auth, submission]);
+
+  const getContent = () => {
+    if (submission.status == "failed") {
+      /*
+      TODO: parse file details for failed alert data
+      Use a useEffect to make the submissionDetails call to continually update
+      */
+      return (
+        <Alert heading="Failed Metadata" type="error">
+          2 Errors were detected
+          <p className="border-1px radius-md border-secondary-light margin-y-2 padding-105">
+            Missing required metadata field, meta_field1.
+          </p>
+          <p className="border-1px radius-md border-secondary-light margin-y-2 padding-105">
+            Metadata field, meta_field2, is set to value3 and does not contain
+            one of the allowed values: [value1, value2]
+          </p>
+        </Alert>
+      );
+    }
+
+    if (submission.status == "completed") {
+      return <ProgressTracker isComplete />;
+    }
+
+    if (submission.status == "processing") {
+      /*
+      TODO: parse file details for stage, current upload amount, total upload amount, etc.
+      Use a useEffect to make the submissionDetails call to continually update
+      */
+      return (
+        <ProgressTracker
+          label="Stage: Uploading"
+          currentAmount={55}
+          totalAmount={256}
+        />
+      );
+    }
+  };
+
+  return (
+    <Modal
+      isOpen={isModalOpen}
+      modalTitle="Submission Details"
+      onClose={handleModalClose}>
+      <ModalBody>
+        <span className="padding-bottom-3">
+          {/* TODO: update cdc-react to allow className on <Pill> so we can avoid this extra span */}
```

Comment:
I'll take care of this in my ticket but leave this TODO for now 👍

Reply:
Sounds good.